Dockerfile

```diff
@@ -1,7 +1,13 @@
 FROM ubuntu:16.04
 
-RUN apt-get update && apt-get install -y curl  libunwind8 libicu55
-RUN curl -LO https://github.com/PowerShell/PowerShell/releases/download/v6.0.0-alpha.9/powershell_6.0.0-alpha.9-1ubuntu1.16.04.1_amd64.deb
-RUN dpkg -i powershell_6.0.0-alpha.9-1ubuntu1.16.04.1_amd64.deb
+ENV POWERSHELL_VERSION="6.0.0-alpha.9"
 
-Entrypoint [ "powershell" ]
+RUN apt-get update && \
+    apt-get install -y curl libunwind8 libicu55 && \
+    curl -LO https://github.com/PowerShell/PowerShell/releases/download/v"$POWERSHELL_VERSION"/powershell_"$POWERSHELL_VERSION"-1ubuntu1.16.04.1_amd64.deb && \
+    dpkg -i powershell_"$POWERSHELL_VERSION"-1ubuntu1.16.04.1_amd64.deb && \
```

Comment:
@garethr, that sounds great! Once there are multiple Dockerfiles, I would place them in tools/docker/<image>.

@RobertKozak while I agree, moving sudo outside the script requires updating the documentation and CI system configuration. It'd be a reasonable change, but I'd posit thay it's outside the scope of cleaning up the Dockerfile.

My next plan (after LinuxCon) is to create Dockerfiles for each supported distribution that support building and packaging from source (a bit more setup than currently in this Dockerfile); so yes, there will be multiple Dockerfiles.